kanidmd/src/lib/idm/authsession.rs

```diff
@@ -301,6 +309,23 @@ impl CredHandler {
                             CredState::Denied(BAD_TOTP_MSG)
                         }
                     }
+                    (AuthCredential::BackupCode(code_chal), _, _, Some(backup_code)) => {
+                        if backup_code.verify(&code_chal) {
+                            pw_mfa.mfa_state = CredVerifyState::Success;
+                            lsecurity!(
+                                au,
+                                "Handler::PasswordMfa -> Result::Continue - BackupCode OK, password -"
+                            );
+                            CredState::Continue(vec![AuthAllowed::Password])
```

Comment:
Consider here adding a `DelayedAction` from kanidmd/src/lib/idm/delayed.rs which will remove code String from account with uuid. See "WebauthnCounterIncrement` for inspiration.